File: data/data.code.lua.client/ReqSceneEnter.lua
module("ReqSceneEnter", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.door_id = 0
end


function encode(self)
	local pack = packet.new()
	pack:writeU32(self.door_id)
	return pack:encode(Msg.P_REQ_SCENE_ENTER)
end

return ReqSceneEnter

File: examples/golang/client/proto/req_scene_enter.go
package proto

import (
	"packet"
)

type ReqSceneEnter struct {
	DoorId                   uint32
}

func (this *ReqSceneEnter) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.DoorId)

	return pack.Encode(P_REQ_SCENE_ENTER)
}
